sdk/lib/rtl/heap.c

```diff
@@ -426,6 +434,9 @@ RtlpRemoveFreeBlock(PHEAP Heap,
     SIZE_T Result, RealSize;
     ULONG HintIndex;
 
+    if (FreeEntry->Size == 0)
```

Comment:
Can you explain why this check is needed and what it means if Size is 0? Please add a comment explaining this, because it looks a bit fishy.

Reply:
This was causing problems when FreeEntry->Size == 0 before the handling for segments > MAXUSHORT was fixed. It may not be needed now. I have added a comment stating this. Thanks.